pkgs/by-name/pa/paup/package.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  autoPatchelfHook,
+  curl,
+  gfortran,
+  zlib,
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "paup";
+  version = "4.0a168";
+
+  src = fetchurl {
```

Comment:
How about using `fetchzip`?

Reply:
This isn't a tarball. It's literally a gzipped executable.